### PR TITLE
The SIMD types must use the struct code path - Fixes issue 19022

### PR DESCRIPTION
**Pull request by briansull**
The SIMD types needed to use the struct code path when we are building
the zero-initializations in fgMorphRecursiveFastTailCallIntoLoop

### CODE REVIEW

File: tests/src/JIT/Regression/JitBlue/GitHub_19022/GitHub_19022.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    class ItemRunner
+    {
+        public ItemRunner()
+        {
+            for (int i = 0; i < _Pool.Length; ++i) { _Pool[i] = new Item(); }
+        }
+
+        private const float _LenghtZ = 1000.0f;
+
+        private static readonly Vector3 _Start = new Vector3(0.0f, -1021.7f, -3451.3f);
```
Comment (tannergooding) [inline tests/src/JIT/Regression/JitBlue/GitHub_19022/GitHub_19022.cs]: Should we also have tests for `Vector128<T>` and `Vector256<T>`, which are similar to, but sometimes handled separately from `Vector2`/`Vector3`/`Vector4`?

Comment (briansull) [reply tests/src/JIT/Regression/JitBlue/GitHub_19022/GitHub_19022.cs]: I don't think that adding more tests would add any additional code-execution coverage here.

Comment (briansull) [reply tests/src/JIT/Regression/JitBlue/GitHub_19022/GitHub_19022.cs]: OSX test failure:
`13:35:25 FAILED   - JIT/HardwareIntrinsics/X86/Avx/Avx_ro/Avx_ro.sh`
Is this a known issue?

Comment (tannergooding) [reply tests/src/JIT/Regression/JitBlue/GitHub_19022/GitHub_19022.cs]: Yes. That is: https://github.com/dotnet/coreclr/issues/19001, which @fiigii was looking at.

**Comment by briansull** [comment]
@dotnet/jit-contrib PTAL

**Comment by vpenades** [comment]
Hi

@briansull I was wondering when this patch will reach end users...  so far we've hacked our app to force running on x86, but it's something we would like to remove ASAP.

@AndyAyersMS we're using Vector SIMD heavily, we began finding compiler issues two years ago, but the fact there's still bugs it's not very encouraging; I talked with developers of other projects that do heavy use of Vectors  [ImageSharp](https://github.com/SixLabors/ImageSharp) , [BEPU Physics](https://github.com/bepu/bepuphysics2) and complain about similar issues.

**Comment by AndyAyersMS** [comment]
@vpenades agree with Tanner here -- If you or others are having problems with SIMD please make sure we know.

**Comment by vpenades** [comment]
@tannergooding the first issue I hit was reported through unofficial channels, at the time I didn't even have a github account. But here's an explanation I gave to someone that also hit that one: https://github.com/SixLabors/ImageSharp/issues/198#issuecomment-298054051

I've been following and contributing to the aforementioned projects, and quite regularly, they've been hitting SIMD and Vector issues, for both NetCore and Xamarin. I guess they've been reported, specially from people of ImageSharp, but I am not specifically tracking these issues.

**Comment by tannergooding** [comment]
@vpenades, thanks for the info.

If you do see any other issues related to SIMD/Vector, it would be useful to point them this way. We can't fix the issues if we don't know about them, and there is way too much code on GitHub to monitor it all ourselves 😄

**Comment by CarolEidt** [comment]
And to "point them this way", you can file issues on GitHub, and mention @eerhardt @tannergooding @AndyAyersMS and/or me (@caroleidt)

**Comment by JimBobSquarePants** [comment]
We've found another `Vector4` issue while developing ImageSharp it seems but are going to have a hard time reducing the code to isolate the underlying cause, I'm hoping it's something you can identify through experience. 

https://github.com/SixLabors/ImageSharp/pull/686/commits/c596fb13fd1a80cacf89c3a17768b606fc20eb99#diff-ab4e2b3561a51a4c90bba84008915ff8R28

When using aggressive inlining the values return to any of the generated methods such as the following were wildly incorrect.

https://github.com/SixLabors/ImageSharp/blob/c596fb13fd1a80cacf89c3a17768b606fc20eb99/src/ImageSharp/PixelFormats/PixelBlenders/PorterDuffFunctions.Generated.cs#L1549

I was able to replicate this consistently with a laptop containing Intel i7-6600U CPU in 64bit on NET Framework 4.6.2 and 4.7.1

**Pull request merged**